### PR TITLE
feat: add more well-known fields

### LINKED ISSUE #1: feat: add more well-known fields by default
Reference: https://angular.io/guide/angular-package-format#legacy-resolution-keys

Related fields: `fesm2020` / `fesm2015` / `esm2020` / `es2020`

I'm glad to raise a PR for it.

### PR DESCRIPTION
**Pull request by JounQin**
close #1

### CODE REVIEW

**Comment by JounQin** [comment]
```sh
/home/runner/work/clean-pkg-json/clean-pkg-json/src/default-keep-properties.ts
Error:   25:1  error  Trailing spaces not allowed  no-trailing-spaces
Error:   59:1  error  Trailing spaces not allowed  no-trailing-spaces

/home/runner/work/clean-pkg-json/clean-pkg-json/src/index.ts
Warning:   33:23  warning  Unexpected any. Specify a different type                            @typescript-eslint/no-explicit-any
Warning:   35:3   warning  Unexpected console statement                                        no-console
Warning:   39:2   warning  Async arrow function has a complexity of 11. Maximum allowed is [10](https://github.com/privatenumber/clean-pkg-json/runs/8005753620?check_suite_focus=true#step:6:11)  complexity
Warning:   84:3   warning  Unexpected console statement                                        no-console
```

I have no idea why CI fails.

File: src/default-keep-properties.ts [hunk 1]
```diff
@@ -44,12 +48,20 @@ export const defaultKeepProperties = [
 	'homepage',
 	'repository',
 	'bugs',
+	'donate', // https://github.com/ThanksApp/donate-spec
```
Comment (privatenumber) [inline src/default-keep-properties.ts]: According to https://github.com/ThanksApp/donate-spec#try-it-out, this is read by https://givethanks.app but it's down

If that's the only app that consumes it, then it probably doesn't need to be in the published package

Also doesn't seem to be very widely adopted based on [GitHub Code Search](https://cs.github.com/?q=%22%5C%22donate%5C%22%3A%22+path%3Apackage.json)

File: src/default-keep-properties.ts [hunk 2]
```diff
@@ -22,6 +22,10 @@ export const defaultKeepProperties = [
 	'browser',
 	'main',
 	'man',
+	
+	// CDN
+	'jsdelivr',
```
Comment (privatenumber) [inline src/default-keep-properties.ts:27]: I know this is a popular one but I couldn't actually find any docs on it. Do you know where it's documented?

File: README.md [hunk 3]
```diff
@@ -71,15 +71,28 @@ By default, these properties are preserved in `package.json`:
...
+#### CDN
+- `jsdelivr`
+- `unpkg`
 
 #### Node.js
 - `type`
 - `exports`
 - `imports`
 
+#### APF
```
Comment (privatenumber) [inline README.md]:
```suggestion
#### Angular Package Format
```

File: README.md [hunk 4]
```diff
@@ -71,15 +71,28 @@ By default, these properties are preserved in `package.json`:
...
+- `fesm2020`
+- `fesm2015`
+- `esm2020`
+- `es2020`
+
 #### TypeScript
 - `types`
+- `typings`
```
Comment (privatenumber) [inline README.md:111]: Add to array

File: src/default-keep-properties.ts [hunk 5]
```diff
@@ -44,12 +48,20 @@ export const defaultKeepProperties = [
 	'homepage',
 	'repository',
 	'bugs',
+	'donate', // https://github.com/ThanksApp/donate-spec
 	'funding',
+	'sponsor', // https://code.visualstudio.com/api/references/extension-manifest
```
Comment (privatenumber) [inline src/default-keep-properties.ts]: Seems there's a lot more properties needed by `vsce`.

Do you mind either removing or adding the rest?

File: src/default-keep-properties.ts [hunk 6]
```diff
@@ -44,12 +48,20 @@ export const defaultKeepProperties = [
...
 
 	// Node.js
 	'type',
 	'exports',
 	'imports',
+	
+	// APF - https://angular.io/guide/angular-package-format#legacy-resolution-keys
```
Comment (privatenumber) [inline src/default-keep-properties.ts]:
```suggestion
	// Angular - https://angular.io/guide/angular-package-format#legacy-resolution-keys
```

**Comment by privatenumber** [comment]
Linting fails because you have a indentation on a line with no text. You can run `pnpm lint --fix` to fix.

File: src/default-keep-properties.ts (hunk 5 above)
Comment (JounQin) [reply src/default-keep-properties.ts]: I'm fine to add.

File: src/default-keep-properties.ts (hunk 1 above)
Comment (JounQin) [reply src/default-keep-properties.ts]: See also https://github.com/feross/thanks/issues/2

I think I'll remove it.

File: src/default-keep-properties.ts (hunk 2 above)
Comment (JounQin) [reply src/default-keep-properties.ts:27]: https://www.jsdelivr.com/features

At the bottom

File: README.md (hunk 4 above)
Comment (JounQin) [reply README.md:111]: ~~Not sure what does `array` mean here.~~

**Comment by JounQin** [comment]
> Linting fails because you have a indentation on a line with no text. You can run `pnpm lint --fix` to fix.

😅 lint-staged and simple-git-hooks/husky should be configured.

**Comment by privatenumber** [comment]
Awesome work, thanks @JounQin

**Pull request merged**